File: app/src/main/java/cc/smartcash/wallet/Utils/NetworkUtil.java
package cc.smartcash.wallet.Utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class NetworkUtil {

    public static int TYPE_WIFI = 1;
    public static int TYPE_MOBILE = 2;
    public static int TYPE_NOT_CONNECTED = 0;

    public static int getConnectivityStatus(Context context) {
        ConnectivityManager cm = (ConnectivityManager) context
                .getSystemService(Context.CONNECTIVITY_SERVICE);

        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
        if (null != activeNetwork) {
            if (activeNetwork.getType() == ConnectivityManager.TYPE_WIFI)
                return TYPE_WIFI;

            if (activeNetwork.getType() == ConnectivityManager.TYPE_MOBILE)
                return TYPE_MOBILE;
        }
        return TYPE_NOT_CONNECTED;
    }

    public static String getConnectivityStatusString(Context context) {
        int conn = NetworkUtil.getConnectivityStatus(context);
        String status = null;

        String internet = " | " + (isInternetAvailable() ? "Internet enabled" : "Internet disabled");

        if (conn == NetworkUtil.TYPE_WIFI) {
            status = "Wifi enabled | " + internet;
        } else if (conn == NetworkUtil.TYPE_MOBILE) {
            status = "Mobile data enabled" + internet;
        } else if (conn == NetworkUtil.TYPE_NOT_CONNECTED) {
            status = "Not connected to Internet" + internet;
        }
        return status;
    }

    public static boolean getInternetStatus(Context context) {
        int conn = NetworkUtil.getConnectivityStatus(context);
        boolean status = false;
        boolean isInternetAvailable = isInternetAvailable();

        if (!isInternetAvailable) return false;

        if (conn == NetworkUtil.TYPE_WIFI && isInternetAvailable) {
            status = true;
        } else if (conn == NetworkUtil.TYPE_MOBILE && isInternetAvailable) {
            status = true;
        } else if (conn == NetworkUtil.TYPE_NOT_CONNECTED) {
            status = false;
        }
        return status;
    }

    public static boolean isInternetAvailable() {
        return true;
    }
}

File: app/src/main/java/cc/smartcash/wallet/Utils/NetworkUtil.kt
package cc.smartcash.wallet.Utils

import android.content.Context
import android.net.ConnectivityManager

object NetworkUtil {

    private var TYPE_WIFI = 1
    private var TYPE_MOBILE = 2
    private var TYPE_NOT_CONNECTED = 0

    private val isInternetAvailable: Boolean
        get() = true

    private fun getConnectivityStatus(context: Context): Int {
        val cm = context
                .getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager

        val activeNetwork = cm.activeNetworkInfo
        if (null != activeNetwork) {
            if (activeNetwork.type == ConnectivityManager.TYPE_WIFI)
                return TYPE_WIFI

            if (activeNetwork.type == ConnectivityManager.TYPE_MOBILE)
                return TYPE_MOBILE
        }
        return TYPE_NOT_CONNECTED
    }

    fun getConnectivityStatusString(context: Context): String? {
        val conn = getConnectivityStatus(context)
        var status: String? = null

        val internet = " | " + if (isInternetAvailable) "Internet enabled" else "Internet disabled"

        when (conn) {
            TYPE_WIFI -> status = "Wifi enabled | $internet"
            TYPE_MOBILE -> status = "Mobile data enabled$internet"
            TYPE_NOT_CONNECTED -> status = "Not connected to Internet$internet"
        }
        return status
    }

    fun getInternetStatus(context: Context): Boolean {
        val conn = getConnectivityStatus(context)
        var status = false
        val isInternetAvailable = isInternetAvailable

        if (!isInternetAvailable) return false

        if (conn == TYPE_WIFI && isInternetAvailable) {
            status = true
        } else if (conn == TYPE_MOBILE && isInternetAvailable) {
            status = true
        } else if (conn == TYPE_NOT_CONNECTED) {
            status = false
        }
        return status
    }
}
